src/scripts/main.js

```diff
@@ -1,3 +1,45 @@
 'use strict';
 
-// write code here
+const list = document.body.querySelectorAll('li');
+
+function sortList(items) {
+  const sorted = Array.from(list).sort((a, b) => {
+    return (
+      parseInt(b.dataset.salary.replace(/[,$]/g, '')) -
+      parseInt(a.dataset.salary.replace(/[,$]/g, ''))
+    );
+  });
+
+  const ul = document.querySelector('ul');
+
+  ul.innerHTML = '';
+
+  sorted.forEach((item) => {
+    ul.appendChild(item);
+  });
+}
+
+sortList(list);
+
+function getEmployees(employees) {
+  const employee = [];
+
+  for (const person of employees) {
+    const obj = {};
+
+    obj.name = person.textContent.trim();
+    obj.position = person.getAttribute('data-position');
+
+    obj.salary = parseInt(
+      person.getAttribute('data-salary').replace(/[,$]/g, ''),
+      10,
+    );
+    obj.age = parseInt(person.getAttribute('data-age'));
```

Comment:
Consider adding a radix parameter to the `parseInt` function for the `age` attribute to ensure consistent parsing. For example: `parseInt(person.getAttribute('data-age'), 10);`